presto-main/src/main/java/com/facebook/presto/operator/scalar/ScalarFunctionImplementation.java

```diff
@@ -106,6 +107,8 @@ public boolean isDeterministic()
         private final List<ArgumentProperty> argumentProperties;
         private final MethodHandle methodHandle;
         private final Optional<MethodHandle> instanceFactory;
+        private final boolean hasSession;
+        private final List<NullConvention> nullConventionList;
```

Comment:
It appears like you want to precompute it and store it here, but you never used it.

This no longer matters given my earlier comments.